// modalize/js/modalize.js
var React = require('react')
  , helpers = require('./helpers.js')
  , Modal = require('./modal.js');

function modalize(Component) {

  var Modalized = React.createClass({displayName: "Modalized",
    statics: {
      show: function (props, container) {
        props.modalVisible = true;
        helpers.show(Modalized, props, container);
      },
      shower: function (props, container) {
        return function () {
          Modalized.show(props, container);
        }
      }
    },
    getInitialState: function(){
      return {
        visible: this.props.modalVisible
      }
    },
    componentWillReceiveProps: function(newProps){
      this.setState({visible: newProps.modalVisible});
    },
    close: function (accepted, payload) {
      /* Hide modal here */
      this.setState({visible: false});
      this.props.modalOnClose(accepted, payload);
    },
    onClose: function(accepted, payload){
      this.setState({visible: false});
      this.props.modalOnClose(accepted, payload);
    },
    render: function () {
      var $__0=
        
        
        
        
        
        
           this.props,modalTitle=$__0.modalTitle,modalVisible=$__0.modalVisible,modalAcceptLabel=$__0.modalAcceptLabel,modalDeclineLabel=$__0.modalDeclineLabel,modalMods=$__0.modalMods,modalOnClose=$__0.modalOnClose,props=(function(source, exclusion) {var rest = {};var hasOwn = Object.prototype.hasOwnProperty;if (source == null) {throw new TypeError();}for (var key in source) {if (hasOwn.call(source, key) && !hasOwn.call(exclusion, key)) {rest[key] = source[key];}}return rest;})($__0,{modalTitle:1,modalVisible:1,modalAcceptLabel:1,modalDeclineLabel:1,modalMods:1,modalOnClose:1});
      props.modalized = true;
      props.close = this.close;

      return (
        React.createElement(Modal, {title: modalTitle, 
               visible: this.state.visible, 
               acceptLabel: modalAcceptLabel, 
               declineLabel: modalDeclineLabel, 
               mods: modalMods, 
               onClose: this.onClose}, 
          React.createElement(Component, React.__spread({},  props), 
            this.props.children
          )
        )
      );
    }
  });

  return Modalized;
}

module.exports = modalize;
