// modalize/jsx/modal.jsx
var React = require('react');

var Modal = React.createClass({
  getDefaultProps: function () {
    return {
      title: null,
      visible: false,
      acceptLabel: null,
      declineLabel: null,
      mods: 'modal-window-bottom',
      onClose: function (accepted, payload) {
      }
    };
  },
  renderHeader: function () {
    if (!this.props.title) {
      return null;
    } else {
      return (
        <div className="modal-header">
          <h3>{this.props.title}</h3>
        </div>
      );
    }
  },
  renderContent: function () {
    return (
      <div className="modal-content-wrap">
        {this.props.children}
      </div>
    );
  },
  onAccept: function (e) {
    this.props.onClose(true, null);
  },
  onDecline: function (e) {
    this.props.onClose(false, null);
  },
  onInnerFormSubmit(e){
    this.props.onClose(true, e.target);
  },
  onInnerFormReset(e){
    this.props.onClose(false, null);
  },
  renderFooter: function () {
    var acceptBtn = null, declineBtn = null;
    if (this.props.acceptLabel) {
      acceptBtn = (
        <button onClick={this.onAccept}>
          {this.props.acceptLabel}
        </button>
      );
    }
    if (this.props.declineLabel) {
      declineBtn = (
        <button onClick={this.onDecline}>
          {this.props.declineLabel}
        </button>
      );
    }
    if (acceptBtn || declineBtn) {
      return (
        <div className="modal-footer">
          {declineBtn}
          {acceptBtn}
        </div>
      );
    } else {
      return null;
    }
  },
  render: function () {
    if (this.props.visible) {
      return (
        <div className="modalize-modal-wrap">
          <div className="modal"
               onSubmit={this.onInnerFormSubmit}
               onReset={this.onInnerFormReset}>
            {this.renderHeader()}
            {this.renderContent()}
            {this.renderFooter()}
          </div>
        </div>
      )
    } else {
      return null;
    }
  }
});

module.exports = Modal;
